cloudinit/sources/azure/errors.py

```diff
@@ -53,11 +53,7 @@ def __init__(
             self.supporting_data = {}
 
         self.timestamp = datetime.now(timezone.utc)
-
-        try:
-            self.vm_id = identity.query_vm_id()
-        except Exception as id_error:
-            self.vm_id = f"failed to read vm id: {id_error!r}"
+        self.vm_id = vm_id
```

Comment:
Doesn't this change mean that most errors reported won't have a `vm_id` associated with them whereas they did before? Is that ok?

Reply:
The report_failure() method populates it when reporting it.

https://github.com/canonical/cloud-init/pull/6199/files#diff-6eb8d221761d9c607e8b836a443abe188b48b23083d7575a58649f9d40b8bcfbR1391

It feels slightly janky doing it this way.  The better alternative is probably to just make it a required param for `ReportableError.as_encoded_report()`.  I'll do that!

Reply:
What do we think about this?